### ISSUE TITLE
Wrong init file when using SPACEMACSDIR and ~/.spacemacs exists

### ISSUE THREAD

**Issue by bmag** [syl20bnr/spacemacs#6669]
I'm trying to make Emacs (on a Windows machine) work with a Spacemacs installation (.emacs.d) and configuration (.spacemacs.d) that are stored on a USB flash drive. If I have a `~/.spacemacs`, and I use `SPACEMACSDIR` environment variable to specify a directory on the USB drive, then Spacemacs tries to load `~/.spacemacs` instead of `%SPACEMACSDIR%/init.el`. The reason is that `dotspacemacs-filepath` is set wrong. PR incoming. There are no repro steps because I'm in the middle of trying to get it working.

**Comment by d12frosted** [comment]
Hm. Indeed. And I was wondering why it was working for me. And then I found following snippet in my `init.el` file.

```
(setq-default spacemacs-start-directory d12-path/spacemacs-home
              dotspacemacs-filepath d12-path/spacemacs-user-config)
```

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.

**Comment by dijonkitchen** [comment]
Why are these cherry-picked instead of merged into the branch? 

Also, what's the cherry-picked commit so that we can check if it's in master yet or not?
